### PR TITLE
More fixes for the serializer

### LINKED ISSUE #8267: Sanitize the changes dict prior to sending it through pickle
The changes to the executor reporting cause the changes dict to be pickled instead of json serialized, this can lead to problems when un-serializing. 

It should be sanitized (or force pickle to json serialize it)

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Make serializer more safe by switching to json when needed

dis-advantages:
1. doesn't reconstruct objects  (which is why it is applied only at the leaves)

closes #8267 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: mypy-baseline.txt [hunk 1]
```diff
@@ -146,6 +146,10 @@ src/inmanta/execute/runtime.py:0: error: Argument 2 to "assign" of "AssignableNo
 src/inmanta/execute/runtime.py:0: error: Argument 1 to "AttributeException" has incompatible type "Instance"; expected "Locatable"  [arg-type]
 src/inmanta/execute/runtime.py:0: error: Missing type parameters for generic type "ResultVariable"  [type-arg]
 src/inmanta/execute/runtime.py:0: error: Argument 1 to "RuntimeException" has incompatible type "Instance"; expected "Locatable | None"  [arg-type]
+src/inmanta/data/model.py:0: error: Return type "str" of "__getstate__" incompatible with return type "dict[Any, Any]" in supertype "BaseModel"  [override]
+src/inmanta/data/model.py:0: error: Argument 1 of "__setstate__" is incompatible with supertype "BaseModel"; supertype defines the argument type as "dict[Any, Any]"  [override]
```
Comment (wouterdb) [inline mypy-baseline.txt:150]: The typing in typeshed does not cover this case

File: src/inmanta/data/__init__.py [hunk 2]
```diff
@@ -4518,7 +4530,7 @@ class Resource(BaseDocument):
 
     # Methods for backward compatibility
     @property
-    def resource_version_id(self):
+    def resource_version_id(self) -> ResourceVersionIdStr:
```
Comment (wouterdb) [inline src/inmanta/data/__init__.py:4533]: accidental type cleanup

**Review by bartv — APPROVED** [review]

File: src/inmanta/data/model.py [hunk 3]
```diff
@@ -242,6 +243,16 @@ def check_serializable(cls, v: Optional[object]) -> Optional[object]:
                 return str(v)
         return v
 
+    def __getstate__(self) -> str:
+        # make pickle use json to keep leaking stuff
```
Comment (sanderr) [inline src/inmanta/data/model.py:247]:
```suggestion
        # make pickle use json to keep from leaking stuff
```

**Review by sanderr — APPROVED** [review]
Still gives me some minor shivers, but I fail to see a more satisfying solution.

**Pull request closed without merging**